src/devices/video/upd7220.cpp

```diff
@@ -999,58 +985,155 @@ void upd7220_device::draw_rectangle(int x, int y)
 //  draw_char -
 //-------------------------------------------------
 
-void upd7220_device::draw_char(int x, int y)
+void upd7220_device::draw_char()
 {
-	int isize,psize;
-	uint16_t tile_data;
+	const int8_t dir_change[2][4] = {
+		{2, 2, -2, -2},
+		{1, 3, -3, -1}
+	};
+	const uint8_t type = (m_figs.m_figure_type & 0x10) >> 4;
 
-	LOG("uPD7220 char check: %d %d %02x %08x %d %d %02x\n",x,y,m_figs.m_dir,m_ead,m_figs.m_d,m_figs.m_dc,m_figs.m_figure_type);
+	uint16_t pattern;
```

Comment:
AFAICT this is only used three loops deep, doesn’t need to persist across iterations, and isn’t modified after initial assignment.  It can be scoped down and `const` qualified.